src/fluent/watch.ts

```diff
@@ -372,6 +372,9 @@ export class Watcher<T extends GenericClass> {
 
       // Make the actual request
       const response = await fetch(url, { ...opts });
+      response.body.on("socket", socket => {
+        socket.setKeepAlive(true, 1000);
```

Comment:
Not sure if this should inform our opinion here but this would be every second I believe (1000 ms) which is diff from the [k8s javascript client's 30s](https://github.com/kubernetes-client/javascript/blob/master/src/watch.ts#L120) - would be good to understand the network traffic/resource impact of this probably?

Reply:
The way I am reading it is that the second argument is an initial delay, from which to start the [keepalive](https://nodejs.org/api/net.html#socketsetkeepaliveenable-initialdelay) probes, but waiting 30 seconds is a good idea to not immediately probe. 

```js
socket.setKeepAlive([enable][, initialDelay])
```

Reply:
actually i take that back, after reading further i think you are right

```plaintext
Set initialDelay (in milliseconds) to set the delay between the last data packet received and the first keepalive probe. Setting 0 for initialDelay will leave the value unchanged from the default (or previous) setting.
```

Reply:
Reviewing this more understanding it as:
- initial delay after data/healthy connection of whatever we add here
- once we hit that interval, send probes every 1 second
- if a probe responds healthy, back to this interval

So I guess comes down to our tolerance here for an unhealthy connection and probably also how lightweight these probes are.